polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+import csv
+import re
+from pathlib import Path
+
+import chardet
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from pyproj import Transformer
+
+from data_importers.s3wrapper import parse_s3_uri, S3Wrapper
+from data_importers.management.commands.import_eoni import Command as EONI_Importer
+
+from polling_stations.settings.constants.importers import EONIImportScheme
+
+
+def attempt_decode(body: bytes):
+    sample = get_body_sample(body)
+    detection = chardet.detect(sample)
+    encoding = detection["encoding"]
+    if encoding == "utf-16le":
+        return body.decode(encoding), encoding
+    encodings = ["utf-8", "windows-1252", "latin-1"]
+    for encoding in encodings:
+        try:
+            return body.decode(encoding), encoding
+        except UnicodeDecodeError as e:
+            last_exception = e
+            continue
+    raise last_exception
+
+
+def get_body_sample(body: bytes):
+    if b"\r\n" in body:
+        line_sep = b"\r\n"
+    elif b"\n" in body:
+        line_sep = b"\n"
+    else:
+        return body
+    lines = body.split(line_sep)
+    if len(lines) > 20:
+        return line_sep.join(lines[:20])
+    return line_sep.join(lines)
```

Comment:
I think we could make a shared package (that lives in the repo) which both the main WDIV codebase and the lambda trigger both depend on. Similar to how the [common](https://github.com/DemocracyClub/aggregator-api/tree/master/api/common) package works in devs.DC.
That said, I reckon I'd probably rather copy & paste 2 functions for now than shave that yak right at this moment.

Reply:
Also I had to test this to double-check but yes you're right. As long as we've parsed it with the correct encoding, when we write it out it will be converted to utf8 :+1: